### ISSUE TITLE
"diffs didn't match during apply" with google_compute_firewall

### ISSUE THREAD

**Issue by julienvey** [hashicorp/terraform#1073]
Tested with 0.3.7 and last version from master

The following tf file causes the `diffs didn't match during apply` error when running `terraform apply`
```
resource "google_compute_network" "subnet_1" {
    name = "subnet-test"
    ipv4_range = "10.0.0.0/24"
}

resource "google_compute_firewall" "test" {
  name = "test-1080"
  network = "${google_compute_network.subnet_1.name}"
  source_ranges = ["${google_compute_address.test_addr.address}/32"]
  allow {
    protocol = "tcp"
  }
}

resource "google_compute_address" "test_addr" {
    name = "test-addr"
}
```
The full log of the command http://pastebin.com/RgncfeZU

Running `terraform apply` a second time works well http://pastebin.com/16PdNUFL

However, if I remove the dependency between `google_compute_firewall` and `google_compute_address`, there is no error

```
resource "google_compute_network" "subnet_1" {
    name = "subnet-test"
    ipv4_range = "10.0.0.0/24"
}

resource "google_compute_firewall" "test" {
  name = "test-1080"
  network = "${google_compute_network.subnet_1.name}"
  source_ranges = ["1.1.1.1/32"]
  allow {
    protocol = "tcp"
  }
}
```
The log of the command http://pastebin.com/4PEuWdKq

**Comment by sparkprime** [comment]
First impression is this looks very similar to https://github.com/hashicorp/terraform/issues/634

**Comment by sparkprime** [comment]
@mitchellh thoughts?

**Comment by sparkprime** [comment]
There is also no error if you remove the dependency on the network (by using the "default" network instead)

**Comment by mitchellh** [comment]
I haven't had a chance to look at this yet but its certainly a core issue. Tagged accordingly and we'll make sure to take a look very soon.

**Comment by mitchellh** [comment]
Reproduced. 

Personal notes so far: The diff isn't properly detecting the set as computed. It is showing up like this (important line): `source_ranges.1902308426:  "" => "74D93920-ED26-11E3-AC10-0800200C9A66/32"`. This should show up as `source_ranges.1902308426: "" => "<computed>"` or something CLOSE to that.

**Comment by mitchellh** [comment]
Fixed by #1082 once merged.
